src/interactions/slashCommands/mixins/CommandChannelOptionBase.ts

```diff
@@ -16,7 +16,13 @@ const allowedChannelTypes = [
 	ChannelType.GuildPrivateThread,
 ];
 
-const channelTypePredicate = ow.number.oneOf(allowedChannelTypes);
+const channelTypePredicate = z.union(
+	allowedChannelTypes.map((type) => z.literal(type)) as [
```

Comment:
This cast feels.. Confusing. Can't you just `as ZodLiteral<ChannelType>[]`?

Reply:
Apparently, not:

https://github.com/colinhacks/zod/blob/master/src/types.ts#L1778-L1779
https://github.com/colinhacks/zod/blob/master/src/types.ts#L1785-L1789
https://github.com/colinhacks/zod/blob/master/src/types.ts#L1867-L1870

Reply:
`ZodLiteral<ChannelType>[]` could be an empty array as far as TS knows, so it won't accept it for an array type that requires at least 2 elements.